src/Configuration/src/Kubernetes.ServiceBinding/PostProcessors.cs

```diff
@@ -0,0 +1,704 @@
+                mapper.MapFromTo("cluster.max-redirects", "clusterMaxRedirects");
+                mapper.MapFromTo("cluster.nodes", "clusterNodes");
+                mapper.MapFromTo("database", "database");
+                mapper.MapFromTo("host", "host");
+                mapper.MapFromTo("password", "password");
+                mapper.MapFromTo("port", "port");
+                mapper.MapFromTo("sentinel.master", "sentinelMaster");
+                mapper.MapFromTo("sentinel.nodes", "sentinelNodes");
+                mapper.MapFromTo("ssl", "ssl");
+                mapper.MapFromTo("url", "url");
+            });
+    }
+}
+
+internal sealed class SapHanaPostProcessor : IConfigurationPostProcessor
+{
+    internal const string BindingTypeKey = "hana";
+
+    public void PostProcessConfiguration(PostProcessorConfigurationProvider provider, IDictionary<string, string> configData)
+    {
+        if (!provider.IsBindingTypeEnabled(BindingTypeKey))
+        {
+            return;
+        }
+
+        configData.Filter(ServiceBindingConfigurationProvider.KubernetesBindingsPrefix, ServiceBindingConfigurationProvider.TypeKey, BindingTypeKey).ForEach(
+            bindingNameKey =>
+            {
+                // Spring -> spring.datasource....
+                // Steeltoe -> steeltoe:hana:binding-name:....
+                var mapper = new ServiceBindingMapper(configData, bindingNameKey, "steeltoe", BindingTypeKey, ConfigurationPath.GetSectionKey(bindingNameKey));
+                mapper.MapFromTo("username", "username");
+                mapper.MapFromTo("password", "password");
+                mapper.MapFromTo("host", "host");
+                mapper.MapFromTo("port", "port");
+                mapper.MapFromTo("database", "database");
+
+                // Spring indicates this takes precedence over above
+                mapper.MapFromTo("jdbc-url", "jdbcUrl");
+
+                // Note: Spring also adds spring.r2dbc.... properties as well
+            });
+    }
+}
+
+internal sealed class SpringSecurityOAuth2PostProcessor : IConfigurationPostProcessor
+{
+    internal const string BindingTypeKey = "oauth2";
+
+    public void PostProcessConfiguration(PostProcessorConfigurationProvider provider, IDictionary<string, string> configData)
+    {
+        if (!provider.IsBindingTypeEnabled(BindingTypeKey))
+        {
+            return;
+        }
+
+        configData.Filter(ServiceBindingConfigurationProvider.KubernetesBindingsPrefix, ServiceBindingConfigurationProvider.TypeKey, BindingTypeKey).ForEach(
+            bindingNameKey =>
+            {
+                // Spring -> spring.security.oauth2.client....
+                // Steeltoe -> steeltoe:oauth2:binding-name:....
+                var mapper = new ServiceBindingMapper(configData, bindingNameKey, "steeltoe", "oauth2", ConfigurationPath.GetSectionKey(bindingNameKey));
+                string provider = mapper.BindingProvider;
```

Comment:
`provider` is already used on line 512, maybe this should be bindingProvider?

Reply:
Changed